src/ir/type-updating.cpp

```diff
@@ -136,8 +148,10 @@ void GlobalTypeRewriter::update() {
         return type;
       }
       if (type.isFunction() || type.isData()) {
-        assert(oldToNewTypes.count(type));
-        return oldToNewTypes[type];
+        auto iter = oldToNewTypes.find(type);
```

Comment:
Is the condition `type.isFunction() || type.isData()` doing anything here? I don't think that can ever be false, unless I'm missing something.

Reply:
Good question, I'm not sure. Perhaps it guards against a null type? Or maybe if we added `isRef()` on heap types (and not just types) we could use that here?

Reply:
Can we just remove it, though?

Reply:
Sorry, I missed your point earlier. Yes, I think that's right, now that the assert is gone we can remove the if too. I'll open a PR.